### DISCUSSION TITLE
Contributing guidelines are a bit confusing

### DISCUSSION THREAD

**Discussion by pawamoy** [pdm-project/pdm#1711] [General]
The CONTRIBUTING docs tell to install dependencies in a venv (not showing how to actually create one) with this:

```bash
python -m pip install -U "pip>=21.3"
python -m pip install -e .
```

But that doesn't actually install the dev-dependencies.

So the next command fails: `pdm run test`.

> You are free to create a virtualenv with either venv module or virtualenv tool for the development. If you are doing so, you may also need to set pdm config python.use_venv true after installation is done.

It should say "must" and not "may", as this is clearly required. Showing how to do so would be nice as well:

```
pdm config -l python.use_venv true

# user probably doesn't know where PDM will create the venv if not in the project
pdm config -l venv.in_project true
```

Then, the docs should say to install the dev-deps with `pdm install`.

---

So, in the end:

```
python -m venv .venv
. .venv/bin/activate
python -m pip install -U pip
python -m pip install -e .
pdm config -l python.use_venv true
pdm config -l venv.in_project true
pdm install
pipx install pre-commit
pre-commit install
pdm run lint
pdm run test
```

Or just this if PDM and pre-commit are already installed:

```
python -m venv .venv
. .venv/bin/activate
python -m pip install -U pip
pdm config -l python.use_venv true
pdm config -l venv.in_project true
pdm install
pre-commit install
pdm run lint
pdm run test
```

**Comment by frostming** [comment]
PR welcomed.

Regarding this:
> ```
> pdm config -l python.use_venv true
> 
> # user probably doesn't know where PDM will create the venv if not in the project
> pdm config -l venv.in_project true
> ```
Since the config values for these two items are both True, we don't need to bother with the details. Instead, one sentence is fine:

>you must also make sure the venv mode is on:
>
> ```bash
> pdm config -l python use_venv true
> ```
